src/libraries/Microsoft.PowerFx.Core/Public/Values/CompileTimeTypeWrapperRecordValue.cs

```diff
@@ -24,5 +27,44 @@ private CompileTimeTypeWrapperRecordValue(RecordType type, RecordValue inner)
             : base(IRContext.NotInSource(type), inner.Fields)
         {
         }
+
+        protected override bool TryGetField(FormulaType fieldType, string fieldName, out FormulaValue result)
```

Comment:
>TryGetField [](http://example.com/codeflow?start=32&length=11)

Are these extra virtuals only required in the If() case where we *reduce* the type? (eg, have an intersection where the resulting type is smaller, and hence TryGetField needs to check if it should "block" a field?)

Ie, in Table() where we do a union, would we ever need these? My concern is the potential cost of the extra lookup. 
Either way - if it is a problem, I guess we'll find it in a profiler.

Reply:
No this is only used for intersection.